src/main/scala/io/iohk/ethereum/consensus/validators/std/StdOmmersValidator.scala

```diff
@@ -66,16 +66,22 @@ class StdOmmersValidator(blockchainConfig: BlockchainConfig, blockHeaderValidato
   /** Validates that each ommer's header is valid based on validations stated in section 11.1 of the YP
     *
     * @param ommers               the list of ommers to validate
-    * @param getBlockHeaderByHash function to obtain ommers' parents
-    *
-    * @return [[OmmersValidator.OmmersValid]] if valid, an [[OmmersValidator.OmmersError.OmmersNotValidError]] otherwise
+    * @param getBlockParentsHeaderByHash function to obtain ommers' parents
+    * @return [[OmmersValidator.OmmersValid]] if valid, an [[OmmersValidator.OmmersError.OmmersHeaderError]] otherwise
     */
   private def validateOmmersHeaders(
       ommers: Seq[BlockHeader],
-      getBlockHeaderByHash: GetBlockHeaderByHash
+      getBlockParentsHeaderByHash: GetBlockHeaderByHash
   ): Either[OmmersError, OmmersValid] = {
-    if (ommers.forall(blockHeaderValidator.validate(_, getBlockHeaderByHash).isRight)) Right(OmmersValid)
-    else Left(OmmersNotValidError)
+    val validationsResult = ommers.map(blockHeaderValidator.validate(_, getBlockParentsHeaderByHash))
+
+    if (validationsResult.forall(_.isRight)) Right(OmmersValid)
+    else {
+      validationsResult.filterNot(_.isRight) match {
+        case Left(headerError) :: _ => Left(OmmersHeaderError(Some(headerError)))
```

Comment:
This will never match because `validationsResult.filterNot(_.isRight)` returns a `Seq`.

Reply:
Yes, and I have deconstructed that in a head and tail, and there's a test proving it works https://github.com/input-output-hk/mantis/blob/feature/ETCM-709-improve-ommers-validation/src/test/scala/io/iohk/ethereum/consensus/validators/std/StdOmmersValidatorSpec.scala#L31-L39
There's never more than one error reported here

Reply:
> here's never more than one error reported here

It's calling `blockHeaderValidator.validate` for every ommer - so why can't there be more than one error?

Reply:
For a couple of reason, the main one was to keep the method signature and the second is because as soon an error is found the block validation is stopped, in general, and I saw no reason to change that

Reply:
Indeed, you are right, it can return more than one error and I can group them all in a list.
I think I was following the approach of stopping the validation at the first error found. I will adjust it to be a list, meaning `case class OmmersHeaderError(error: List[BlockHeaderError])`